pkg-manager/plugin-commands-installation/src/add.ts

```diff
@@ -223,6 +223,12 @@ export async function handler (
       ...(opts.rootProjectManifest.pnpm.onlyBuiltDependencies ?? []),
       ...opts.allowBuild,
     ])).sort((a, b) => a.localeCompare(b))
+    if (opts.rootProjectManifest.pnpm.ignoredBuiltDependencies?.length) {
+      opts.rootProjectManifest.pnpm.ignoredBuiltDependencies = opts.rootProjectManifest.pnpm.ignoredBuiltDependencies.filter((dep) => !opts.allowBuild?.includes(dep))
+      if (!opts.rootProjectManifest.pnpm.ignoredBuiltDependencies.length) {
+        delete opts.rootProjectManifest.pnpm.ignoredBuiltDependencies
+      }
+    }
```

Comment:
After thinking about it a bit more. I am not sure, maybe the right solution would be to throw an error in this case.

Reply:
Should we throw an error directly and not replace the content? Or should we give a prompt if there is an overlap and let the user decide whether to replace or cancel the content?

Reply:
No, I don't think a prompt is needed. Just throw an error.

Reply:
Okay.